public/js/components/SortingBar.jsx

```diff
@@ -9,18 +9,36 @@ export default class SortingBar extends React.Component {
     ArticleActions.fetchArticles(sort)
   }
 
+  constructor() {
+    super()
+    this.state = {
+      newActive: null,
+      topActive: null,
+    }
+  }
+
+  setNewActive() {
+    this.setState({ newActive: true, topActive: null })
+  }
+
+  setTopActive() {
+    this.setState({ newActive: null, topActive: true })
+  }
+
   render() {
     return (
       <ButtonGroup>
         <Button
-          onClick={() => SortingBar.changeSort('NEW')}
+          onClick={() => { SortingBar.changeSort('NEW'); this.setNewActive() }}
           bsStyle="info"
           bsSize="small"
+          active={this.state.newActive}
         >Newest</Button>
         <Button
-          onClick={() => SortingBar.changeSort('TOP')}
+          onClick={() => { SortingBar.changeSort('TOP'); this.setTopActive() }}
```

Comment:
but basically: 
You have to be careful about the meaning of this in JSX callbacks. In JavaScript, class methods are not bound by default. If you forget to bind this.handleClick and pass it to onClick, this will be undefined when the function is actually called.

This is not React-specific behavior; it is a part of how functions work in JavaScript. Generally, if you refer to a method without () after it, such as onClick={this.handleClick}, you should bind that method.